pinot-core/src/test/java/org/apache/pinot/core/common/datatable/DataTableSerDeTest.java

```diff
@@ -123,36 +123,142 @@ public void testException(int dataTableVersion)
   }
 
   @Test(dataProvider = "versionProvider")
-  public void testEmptyStrings(int dataTableVersion)
+  public void testEmptyValues(int dataTableVersion)
       throws IOException {
     DataTableBuilderFactory.setDataTableVersion(dataTableVersion);
     String emptyString = StringUtils.EMPTY;
     String[] emptyStringArray = {StringUtils.EMPTY};
+    ByteArray emptyBytes = new ByteArray(new byte[0]);
+    for (int numRows = 0; numRows < NUM_ROWS; numRows++) {
+      testEmptyValues(new DataSchema(new String[]{"STR_SV", "STR_MV"}, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.STRING, DataSchema.ColumnDataType.STRING_ARRAY
+      }), numRows, new Object[]{emptyString, emptyStringArray});
+
+      testEmptyValues(
+          new DataSchema(new String[]{"STR_SV"}, new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.STRING}),
+          numRows, new Object[]{emptyString});
+
+      testEmptyValues(new DataSchema(new String[]{"STR_MV"},
+              new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.STRING_ARRAY}), numRows,
+          new Object[]{emptyStringArray});
+
+      testEmptyValues(
+          new DataSchema(new String[]{"BYTES"}, new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.BYTES}),
+          numRows, new Object[]{emptyBytes});
+
+      testEmptyValues(
+          new DataSchema(new String[]{"BOOL_ARR"},
+              new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.BOOLEAN_ARRAY}),
+          numRows, new Object[]{new int[]{}});
+
+      testEmptyValues(
```

Comment:
why is this one initialized with a `{1}`?

Reply:
Wanna test array with 0 element and 1 element with default value.

Reply:
this should be `new int[]{0}` for boolean default